src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -173,8 +173,13 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
       _uuidFactory = configs.getConfiguredInstance(LiKafkaProducerConfig.UUID_FACTORY_CLASS_CONFIG, UUIDFactory.class);
       _messageSplitter = new MessageSplitterImpl(_maxMessageSegmentSize, segmentSerializer, _uuidFactory);
       // Instantiate auditor if necessary
-      _auditor = auditor != null ? auditor
-          : configs.getConfiguredInstance(LiKafkaProducerConfig.AUDITOR_CLASS_CONFIG, Auditor.class);
+      LiKafkaProducerConfig auditorConfigs = configs.configsWithCurrentProducer(_producer);
```

Comment:
This will potentially print the producer configurations twice in the log, which could be a little annoying. I think it might be cleaner to just override the `getConfiguredInstance()` method.